### ISSUE TITLE
sign compare warning in Py

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#1198]
```
/root/repo/packages/openpmd-api/build/openpmd-api-0.14.4/src/binding/python/RecordComponent.cpp:957:77: warning: comparison of integers of different signs: 'pybind11::ssize_t' (aka 'long') and 'unsigned int' [-Wsign-compare]
            if( offset_in.size() == 1u && offset_in.at(0) == 0u && a.ndim() > 1u )
                                                                   ~~~~~~~~ ^ ~~
```
